pkg/audit/containment_test.go

```diff
@@ -6,16 +6,212 @@ package audit_test
+	"github.com/golang/protobuf/ptypes/timestamp"
 	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
 
+	"storj.io/storj/internal/memory"
 	"storj.io/storj/internal/testcontext"
 	"storj.io/storj/internal/testplanet"
 	"storj.io/storj/pkg/audit"
+	"storj.io/storj/pkg/auth/signing"
+	"storj.io/storj/pkg/pb"
 	"storj.io/storj/pkg/pkcrypto"
 	"storj.io/storj/pkg/storj"
+	"storj.io/storj/uplink"
 )
 
+func TestReverifySuccess(t *testing.T) {
+	testplanet.Run(t, testplanet.Config{
+		SatelliteCount: 1, StorageNodeCount: 6, UplinkCount: 1,
+	}, func(t *testing.T, ctx *testcontext.Context, planet *testplanet.Planet) {
+
+		// This is a bulky test but all it's doing is:
+		// - uploads random data
+		// - uses the cursor to get a stripe
+		// - creates pending audits for all nodes holding pieces for that stripe
+		//     - the actual shares are downloaded to make sure ExpectedShareHash is correct
+		// - calls reverify on that same stripe
+		// - expects all six storage nodes to be marked as successes in the audit report
+
+		err := planet.Satellites[0].Audit.Service.Close()
+		require.NoError(t, err)
+
+		ul := planet.Uplinks[0]
+		testData := make([]byte, 1*memory.MiB)
+		_, err = rand.Read(testData)
+		require.NoError(t, err)
+
+		err = ul.UploadWithConfig(ctx, planet.Satellites[0], &uplink.RSConfig{
+			MinThreshold:     4,
+			RepairThreshold:  5,
+			SuccessThreshold: 6,
+			MaxThreshold:     6,
+		}, "testbucket", "test/path", testData)
+		require.NoError(t, err)
+
+		metainfo := planet.Satellites[0].Metainfo.Service
+		cursor := audit.NewCursor(metainfo)
+
+		var stripe *audit.Stripe
+		stripe, _, err = cursor.NextStripe(ctx)
+		require.NoError(t, err)
+		require.NotNil(t, stripe)
+
+		overlay := planet.Satellites[0].Overlay.Service
+		transport := planet.Satellites[0].Transport
+		orders := planet.Satellites[0].Orders.Service
+		containment := planet.Satellites[0].DB.Containment()
+		minBytesPerSecond := 128 * memory.B
+		reporter := audit.NewReporter(overlay, containment, 1)
+		verifier := audit.NewVerifier(zap.L(), reporter, transport, overlay, containment, orders, planet.Satellites[0].Identity, minBytesPerSecond)
+		require.NotNil(t, verifier)
+
+		for _, piece := range stripe.Segment.GetRemote().GetRemotePieces() {
+			rootPieceID := stripe.Segment.GetRemote().RootPieceId
+			redundancy := stripe.Segment.GetRemote().GetRedundancy()
+
+			orderLimit, err := signing.SignOrderLimit(signing.SignerFromFullIdentity(planet.Satellites[0].Identity), &pb.OrderLimit2{
+				SerialNumber:    storj.SerialNumber{},
+				SatelliteId:     planet.Satellites[0].ID(),
+				UplinkId:        ul.ID(),
+				StorageNodeId:   piece.NodeId,
+				PieceId:         rootPieceID.Derive(piece.NodeId),
+				Action:          pb.PieceAction_GET_AUDIT,
+				Limit:           int64(redundancy.ErasureShareSize),
+				PieceExpiration: &timestamp.Timestamp{Seconds: time.Now().Unix() + 3000},
+				OrderExpiration: &timestamp.Timestamp{Seconds: time.Now().Unix() + 3000},
+			})
```

Comment:
I think we can simplify some code here by requesting the order limit from the Orders Service instead of replicating the same code here.

Reply:
Did it with https://github.com/storj/storj/pull/2041/commits/bcc49ba344df1a822e33ddf70c2c9eda27fad2ce